### PR TITLE
cmake: add autotest, b2bua, intercom, kaoptions

### PR DESCRIPTION
**Pull request by cspiel1**

### CODE REVIEW

**Comment by cspiel1** [comment]
@sreimers 
This is a marvel to me:
```
make: *** No rule to make target 'build-x86_64/src/rtpext.o', needed by 'baresip'.  Stop.
```
rtpext.c was compiled during `re` build.

Another question:
We want a simple yocto recipe for baresip-apps. Should we add a main CMakeLists.txt?

**Comment by cspiel1** [comment]
I am not sure:
- Should build.yml do still test the old Makefile build?
- If yes, how long will we support the old Makefile?

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+if(USE_OPENSSL)
+  list(APPEND LINKLIBS OpenSSL::SSL OpenSSL::Crypto)
+endif()
+
+if(WIN32)
+  list(APPEND LINKLIBS ws2_32 iphlpapi winmm gdi32 crypt32 strmiids
+      ole32 oleaut32 qwave)
+else()
+  list(APPEND LINKLIBS -lm)
+endif()
```
Comment (sreimers) [inline CMakeLists.txt:200]: I think this section can be removed too. Linking should not necessary for modules.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+# Pre-release identifier, comment out on a release
+# Increment for breaking changes (dev2, dev3...)
+set(PROJECT_VERSION_PRE dev)
+
+if(PROJECT_VERSION_PRE)
+  set(PROJECT_VERSION_FULL ${PROJECT_VERSION}-${PROJECT_VERSION_PRE})
+else()
+  set(PROJECT_VERSION_FULL ${PROJECT_VERSION})
+endif()
```
Comment (sreimers) [inline CMakeLists.txt:28]: I think we can remove the hole versioning stuff here.

**Comment by cspiel1** [comment]
@sreimers questions about the github builds: https://github.com/baresip/baresip-apps/pull/20#issuecomment-1318834538

**Comment by cspiel1** [comment]
This should fix the build:
https://github.com/baresip/baresip/pull/2336

**Comment by cspiel1** [comment]
Okay, so we should first make a PR that switches build.yml to cmake. Maybe I could do this tomorrow.

**Comment by cspiel1** [comment]
Replaced by: https://github.com/baresip/baresip-apps/pull/21

**Pull request closed without merging**